samples/excel/89-preview-apis/chart-axis-formatting.yaml

```diff
@@ -0,0 +1,189 @@
+                axis.displayUnit = "None";
+                axis.scaleType = "Logarithmic";
+                axis.logBase = 100;
+                axis.majorTickMark = "Cross";
+                // Set the product name label location.
+                // In this case, the location is at 100 units.
+                axis.setCrossesAt(100);
+
+                await context.sync();
+            });
+        }
+
+        async function setup() {
+            await Excel.run(async (context) => {
+                let sheet = await OfficeHelpers.ExcelUtilities
+                    .forceCreateSheet(context.workbook, "Sample");
+                let salesTable = sheet.tables.add('A1:E1', true);
+                salesTable.name = "SalesTable";
+
+                salesTable.getHeaderRowRange().values = [["Product", "Qtr1", "Qtr2", "Qtr3", "Qtr4"]];
+
+                salesTable.rows.add(null, [
+                    ["Frames", 5000, 7000, 6544, 4377],
+                    ["Saddles", 400, 323, 276, 651],
+                    ["Brake levers", 12000, 8766, 8456, 9812],
+                    ["Chains", 1550, 1088, 692, 853],
+                    ["Mirrors", 225, 600, 923, 544],
+                    ["Spokes", 6005, 7634, 4589, 8765]
+                ]);
+
+                if (Office.context.requirements.isSetSupported("ExcelApi", 1.7)) {
+                    sheet.getUsedRange().format.autofitColumns();
+                    sheet.getUsedRange().format.autofitRows();
+                }
+
+                createChart(context);
+                displayUnit(context);
+                sheet.activate();
+
+                await context.sync();
+            });
+        }
+
+        async function createChart(context: Excel.RequestContext) {
+            let sheet = context.workbook.worksheets.getItem("Sample");
+            let salesTable = sheet.tables.getItem("SalesTable");
+
+            let dataRange = sheet.getRange("A1:B7");
+            let chart = sheet.charts.add("Line", dataRange, "auto");
+
+            chart.setPosition("A15", "I30");
+            chart.legend.position = "right"
+            chart.legend.format.fill.setSolidColor("white");
+            chart.dataLabels.format.font.size = 15;
+            chart.dataLabels.format.font.color = "black";
+            
+            chart.axes.categoryAxis.majorGridlines.visible = true;
+            chart.axes.categoryAxis.majorTickMark = "None";
+            chart.title.text = "Sales of Bicycle Parts";
+            chart.name = "SalesChart";
+
+            await context.sync();
+        }
+
+        async function displayUnit(context: Excel.RequestContext) {
+            let sheet = context.workbook.worksheets.getItem("Sample");
+
+            let chart = sheet.charts.getItem("SalesChart");
+
+            let axis = chart.axes.valueAxis;
+            axis.displayUnit = "Thousands";
+
+            await context.sync();
```

Comment:
Sorry I didn't see this before, but since the displayUnit is called from within a function that itself has a context.sync, I think the one inside displayUnit is unneeded. I think if you get rid of it (and remove the async keyword from the beginning of displayUnit, it should still work.